### ISSUE TITLE
Installation fails with "Cannot read property 'clientAllowedKeys' of undefined"

### ISSUE THREAD

**Issue by jagthedrummer** [fivetanley/ember-cli-dotenv#54]
```
$ ember --version
ember-cli: 3.4.3
node: 8.10.0
os: darwin x64
```

```
$ ember install ember-cli-dotenv
Yarn: Installed ember-cli-dotenv
installing ember-cli-dotenv
Cannot read property 'clientAllowedKeys' of undefined


Stack Trace and Error Report: /var/folders/hb/jkd1b3q16nl2xk1qx58mgl3m0000gn/T/error.dump.eb0a01676b4da93523e75d8b44ee1844.log
```

The stack trace from the log file listed above:

```
  - stack: TypeError: Cannot read property 'clientAllowedKeys' of undefined
    at Class.config (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli-dotenv/index.js:68:36)
    at addons.reduce (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli/lib/models/project.js:323:31)
    at Array.reduce (<anonymous>)
    at Project.getAddonsConfig (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli/lib/models/project.js:321:24)
    at Project.configWithoutCache (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli/lib/models/project.js:272:31)
    at Project.config (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli/lib/models/project.js:255:21)
    at Class._generateFileMapVariables (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli/lib/models/blueprint.js:792:40)
    at Promise.then.customLocals (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli/lib/models/blueprint.js:825:35)
    at tryCatcher (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:323:19)
    at invokeCallback (/Users/jgreen/projects/remarq-books-ember/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:495:31)
```

**Comment by jagthedrummer** [comment]
Looks like maybe installation didn't _fail_, maybe it just complained about the config file not being there yet. I do see changes to `package.json` and `yarn.lock` and find `ember-cli-dotenv` in `node_modules`.

I added this to `config/dotenv.js` and then tried to re-install and everything was fine.

```
module.exports = function(env) {
  return {
    clientAllowedKeys: [],
    // Fail build when there is missing any of clientAllowedKeys environment variables.
    // By default false.
    failOnMissingKey: false,
  };
};
```

```
$ ember install ember-cli-dotenv
Yarn: Installed ember-cli-dotenv
installing ember-cli-dotenv
Installed addon package.
```

**Comment by shanekoss** [comment]
Above still gave same error for me.  I had install version 2.1.0 and then it worked.  Running ember 2.18

**Comment by Uysim** [comment]
Meet the same problem on ember-cli latest

**Comment by muziejus** [comment]
Ran into the same problem w/ ember-cli 3.9.0-beta.2. @jagthedrummer's fix seems to have worked.

**Comment by SergeAstapov** [comment]
Fixed in 2.2.2
